### ISSUE TITLE
Why Dart VM keep emitting native code running in Debug mode on iOS?

### ISSUE THREAD

**Issue by wqyfavor** [flutter/flutter#57111]
I run Flutter on iOS device in debug mode. I think in this mode Dart code is interpreted without AOT or JIT.

With breakpoints in DartVM, I find that functions in snapshot are compiled lazily at runtime.
But what makes me confused is that why DartVM keep generating native assembly code on iOS when running Flutter demo. Do we have permission to execute the assembly code?

<img width="1263" alt="截屏2020-05-13 下午9 48 42" src="https://user-images.githubusercontent.com/113776/81820723-8db56e00-9563-11ea-8e78-767ac1cb4732.png">

By the way I am trying to figure out how dart_entry works. In `DartEntry::InvokeFunction`, `StubCode::InvokeDartCode().EntryPoint()` is a stub compiled at Dart VM initialization phase. I follow the code and find that this stub is also compiled as assembly code. I am confused about what function is variable `entrypoint` pointing to?

<img width="1357" alt="截屏2020-05-13 下午9 34 53" src="https://user-images.githubusercontent.com/113776/81820975-e4bb4300-9563-11ea-8ee6-2786fbf94349.png">

<img width="1325" alt="截屏2020-05-13 下午9 45 44" src="https://user-images.githubusercontent.com/113776/81820754-95751280-9563-11ea-93be-8f6e4aee6ee6.png">

From native to Dart, I cannot see the exact function name of stacks in DartVM. I tried to use "dladdr((const void *)address, &imageInfo)" to find the image name and symbol name of the address but these addresses are not valid.

<img width="1390" alt="截屏2020-05-13 下午9 38 04" src="https://user-images.githubusercontent.com/113776/81820825-aa51a600-9563-11ea-864b-c832e14338bc.png">

I have three questions. (All I talk about are in iOS debug mode without AOT.)
a) In debug mode on iOS, why Dart VM keep generating native assembly code on iOS? Do we actually use these assembly code?
b) In debug mode on iOS, how DartVM executes instructions from snapshots? I don't think we have permission to execute the JIT assembly code. So there must be some interpreter. Can you provide some source file names which will help me understanding how DartVM interprets instructions on iOS without JIT?
c) From native to Dart, what are these functions without any symbol info. How are they generated, from source or automatically generated? Where are they, in App.framework or Flutter.framework?

Thanks a lot.

**Comment by kangwang1988** [comment]
@mraleph 
I think you have more context and knowledge towards this.

**Comment by mraleph** [comment]
We indeed used to use an interpreter in debug mode (called DBC), but last year we realized that during development one can actually use JIT, because normal iOS codesign restrictions are lifted if the process is ptraced. See https://saagarjha.com/blog/2020/02/23/jailed-just-in-time-compilation-on-ios/  for more details. 

That’s why you see all this new code being generated - we do in fact use JIT.

This should give you the answer to all your three questions, if not - let me know.
